# Projeto individualdo módulo 4
## de Rafael César

### Primeiro passo: Criar a base de dados

Nós vamos começar criando o nosso DataFrame, que vai ser a nossa base de dados para este projeto.

Para isso vamos Importar o Pandas e o Numbpy que são duas bibliotecas do python

1. O Pandas serve para a manipulação de dados comopor exemplo de Series e DataFrames

2. O Numpy serve para operações matemáticas e lógicas e auxilia o Pandas, já que o Pandas foi criado usando o Numpy de base.

- Para usar essas bibliotecas temos que instala-las usando os comandos.

        !pip install 

In [32]:
!pip install numpy
!pip install pandas


[notice] A new release of pip available: 22.3.1 -> 23.2.1
[notice] To update, run: python.exe -m pip install --upgrade pip



[notice] A new release of pip available: 22.3.1 -> 23.2.1
[notice] To update, run: python.exe -m pip install --upgrade pip


Após a intalação temos que importar essa bibliotecas

In [33]:
import numpy as np
import pandas as pd

Em seguida vamos criar essa tabela

![tabela](tabela_print.png)

In [34]:
# Criamos um array com os dados.
tabela_dados = np.array([('Segunda', 6, 3, 5), ('Terça', 7, 2, 4), ('Quarta', 8, 1, 6), ('Quinta', 6, 4, 4), ('Sexta', 7, 3, 5), ('Sábado', 5, 2, 3), ('Domingo', 4,1,2)])
# Transformamos esse array em um DataFrame e nomenado corretamente as colunas.
data = pd.DataFrame(tabela_dados, columns = ['Dia', 'Horas Trabalhadas', 'Bugs Corrigidos', 'Tarefas Concluídas'])

In [35]:
# Corrigimos as Colunas para ter o formato correto de dados
data = data.astype({'Horas Trabalhadas': 'int', 'Bugs Corrigidos': 'int', 'Tarefas Concluídas':'int'})
# Vizualizamos o DataFrame
data

Dia  Horas Trabalhadas  Bugs Corrigidos  Tarefas Concluídas
0  Segunda                  6                3                   5
1    Terça                  7                2                   4
2   Quarta                  8                1                   6
3   Quinta                  6                4                   4
4    Sexta                  7                3                   5
5   Sábado                  5                2                   3
6  Domingo                  4                1                   2

Agora com o Dataframe criado, vamos criar o relatório.

### Segundo Passo: Gerar o relatório.

Para gerar o relatório devemos calcular as seguintes estatísticas: 
1. Total de Horas Trabalhadas
2. Média Diária de Horas Trabalhadas
3. Total de Bugs Corrigidos
4. Média Diária de Bugs Corrigidos
5. Total de Tarefas Concluídas
6. Média Diária de Tarefas Concluídas
7. Produtividade Diária (Tarefas Concluídas por Hora)

1. Total de horas trabalhadas:

In [36]:
total_horas_trabalhadas = data['Horas Trabalhadas'].sum()
total_horas_trabalhadas

43

        Temos um total de 43 horas trabalhadas na semana.

Isso nos faz ter uma noção da quantidade de trabalho está sendo realizada em uma semana. podendo por exemplo gerar uma base para um gráfico Burndown

2. Média diária de horas trabalhadas

In [37]:
media_horas_trabalhadas = data['Horas Trabalhadas'].mean()
media_horas_trabalhadas

6.142857142857143

        Temos uma média de 6.14 de Horas trabalhadas por dia.
A média de horas trabalhadas pode ser usada para saber quais são os dias que mais rendem ao compara-los com a média por exemplo,

3. Total de bugs corrigidos

In [38]:
total_bugs_corrigidos = data['Bugs Corrigidos'].sum()
total_bugs_corrigidos

16

        Foi resolvido um total de 16 bugs nessa semana.
O total de Bugs corrigidos pode nos indicar também a quantidade de bugs que o processo está gerando.

4. Média diária de bugs corrigidos

In [39]:
media_bugs_corrigidos = data['Bugs Corrigidos'].mean()
media_bugs_corrigidos

2.2857142857142856

        A média de bugs resolvida por dia é cerca de 2.28.

5. Total de tarefas concluídas

In [40]:
total_tarefas_concluidas = data['Tarefas Concluídas'].sum()
total_tarefas_concluidas

29

        Temos um total de 29 tarefas concluidas dentro da semana.

6. Média diária de tarefas concluídas

In [41]:
media_tarefas_concluidas = data['Tarefas Concluídas'].mean()
media_tarefas_concluidas

4.142857142857143

        E a média diária de tarefas feitas pela equipe foi 4.14.

7. Produtividade diária (Tarefas concluidas por hora)

In [42]:
produtividade = total_tarefas_concluidas / total_horas_trabalhadas
produtividade

0.6744186046511628

        A produtividade (tarefas por hora) alcançou a marca de 0.67 ou seja uma tarefa concluida a cada 1.48 horas

### Terceiro passo: Tabela com os resultados do relatório.

In [43]:
tabela_relatorio = {'Dados' : ['Horas Trabalhadas', 'Bugs Resolvidos', 'Tarefas Concluídas'],
         'Total da semana': [total_horas_trabalhadas,total_bugs_corrigidos, total_tarefas_concluidas],
         'Média diária': [media_horas_trabalhadas, media_bugs_corrigidos, media_tarefas_concluidas]}
data2 = pd.DataFrame(tabela_relatorio)
data2

Dados  Total da semana  Média diária
0   Horas Trabalhadas               43      6.142857
1     Bugs Resolvidos               16      2.285714
2  Tarefas Concluídas               29      4.142857

### Quarto passo: exportar pro power BI

In [44]:
data.to_csv('../PowerBI/tabela.csv')

![powerBI](powerBI.png)
Gráfico de série temporal com Horas Trabalhadas, Tarefas Concluídas e Bugs Resolvidos.